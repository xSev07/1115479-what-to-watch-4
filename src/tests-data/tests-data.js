export const movies = [
  {
    id: `Hotel-2000`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`
    ],
    duration: 60,
    genre: `Drama`,
    inList: false,
    producer: `Wes Andreson`,
    rating: 8.9,
    reviews: [],
    title: `The Grand Budapest Hotel`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    votes: 240,
    year: 2000,
  },
  {
    id: `Bohemian-2018`,
    actors: [
      `Rami Malek`,
      `Lucy Boynton`,
      `Gwilym Lee`
    ],
    description: [
      `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Музыкальный фестиваль Live Aid (1985)`
    ],
    duration: 134,
    genre: `Drama`,
    inList: false,
    producer: `Bryan Singer`,
    rating: 8.0,
    reviews: [],
    title: `Bohemian Rhapsody`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    votes: 220,
    year: 2018,
  },
  {
    id: `Macbeth-2015`,
    actors: [
      `Michael Fassbender`,
      `Marion Cotillard`,
      `Paddy Considine`
    ],
    description: [
      `From the Academy Award winning producers of The King’s Speech and acclaimed director Justin Kurzel, comes a visceral and visually breath-taking retelling of the classic tale about an ambitious Scottish lord who seizes the throne with the help of his wife. Starring Academy Award nominee Michael Fassbender and Academy Award winner Marion Cotillard, Macbeth is both visually epic and a breathtaking experience.`
    ],
    duration: 113,
    genre: `Documental`,
    inList: false,
    producer: `Justin Kurzel`,
    rating: 6.6,
    reviews: [],
    title: `Macbeth`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    votes: 220,
    year: 2015,
  },
  {
    id: `Aviator-2005`,
    actors: [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Kate Beckinsale`
    ],
    description: [
      `The Aviator is a 2004 American epic biographical drama film directed by Martin Scorsese and written by John Logan. It stars Leonardo DiCaprio as Howard Hughes, Cate Blanchett as Katharine Hepburn, and Kate Beckinsale as Ava Gardner. The supporting cast features Ian Holm, John C. Reilly, Alec Baldwin, Jude Law as Errol Flynn, Gwen Stefani as Jean Harlow, Kelli Garner as Faith Domergue, Matt Ross, Willem Dafoe, Alan Alda, and Edward Herrmann.`,
      `Based on the 1993 non-fiction book Howard Hughes: The Secret Life by Charles Higham, the film depicts the life of Howard Hughes, an aviation pioneer and director of Hell's Angels. The film portrays his life from 1927–1947 during which time Hughes became a successful film producer and an aviation magnate while simultaneously growing more unstable due to severe obsessive–compulsive disorder (OCD).`
    ],
    duration: 170,
    genre: `Thriller`,
    inList: false,
    producer: `Martin Scorsese`,
    rating: 7.5,
    reviews: [],
    title: `Aviator`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    votes: 220,
    year: 2005,
  },
];

export const promoMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014
};
