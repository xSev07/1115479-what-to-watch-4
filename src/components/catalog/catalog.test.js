import React from "react";
import renderer from "react-test-renderer";
import Catalog from "./catalog";
import {genres, movies, storeData} from "../../tests-data/tests-data";
import configureStore from "redux-mock-store";
import {StaticRouter} from "react-router-dom";

const mockStore = configureStore([]);

it(`Should Catalog render correctly`, () => {
  const store = mockStore(storeData);

  const tree = renderer
    .create(
        <StaticRouter>
          <Catalog
            store={store}
            movies={movies}
            genres={genres}
            activeGenre={`all genres`}
            onGenreClick={() => {}}
            onMovieCardClick={() => {}}
          />
        </StaticRouter>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
