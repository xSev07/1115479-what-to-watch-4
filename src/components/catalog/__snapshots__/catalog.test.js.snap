// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Catalog render correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Documental
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Drama
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Thriller
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/Hotel-2000"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          The Grand Budapest Hotel
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/Bohemian-2018"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          Bohemian Rhapsody
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/Macbeth-2015"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          Macbeth
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/Aviator-2005"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          Aviator
        </h3>
      </a>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
